src/Agent/NewRelic/Agent/Extensions/NewRelic.Agent.Extensions/Configuration/IConfiguration.cs

```diff
@@ -91,6 +91,7 @@ public interface IConfiguration
         IEnumerable<string> ExpectedErrorClassesForAgentSettings { get; }
         IDictionary<string, IEnumerable<string>> ExpectedErrorMessagesForAgentSettings { get; }
         string ExpectedErrorStatusCodesForAgentSettings { get; }
+        IDictionary<string, IEnumerable<string>> ExpectedErrorsInfo { get; }
```

Comment:
I feel like there may be an opportunity to encapsulate all of this into a single `ExpectedErrorsConfiguration` class. The class could maintain the notion of:
* The original config specified by the customer for reporting to `agent_settings`.
* The expected class/message config, and
* The expected status code config

This is just an idea to explore in a follow up PR, maybe when you do the work on dealing with expected status codes.

Reply:
I make sense to encapsulate all of these configurations into a single one, but I would like keep it that way for now as I want to see how the expected status code implementation work will impact this.